scripts/npm-audit.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+set -eou pipefail
+
+export NVM_DIR=$HOME/.nvm;
+# shellcheck source=/dev/null
+source "$NVM_DIR/nvm.sh";
+# default to project root node version
+nvm use
+
+for dir in ./{examples,packages}/*; do
+  if [ -d "$dir" ]; then
+  (
+    cd "$dir"
+
+    # use node version if .nvmrc exists
+    if [ -f .nvmrc ]; then
+      nvm use
+    else # use node 16 / npm 7+ if lockfile version is 2
+      if [ "$(jq -r '.lockfileVersion' package-lock.json)" = "2" ]; then
+        nvm use 16
+      fi
+    fi
+
+    # use yarn if yarn.lock exists
+    if [ -f yarn.lock ]; then
+      echo "Installing dependencies in $dir"
+      yarn install || exit 0
+
+      echo "Running 'yarn audit fix' in $dir"
+      yarn audit fix || exit 0
```

Comment:
this only works for `yarn v1`:

v1: https://classic.yarnpkg.com/en/docs/cli/audit#search
v3: https://yarnpkg.com/cli/npm/audit

We have both being used in this mono repo. I handled this issue with a `yarn --v` check in a zx script:

https://github.com/BuilderIO/builder/blob/0425af5b64a9076c7c40cc1d118f3d98a1139d47/scripts/update-npm-dependency.mjs#L72-L80